lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/audio_output.rb

```diff
@@ -0,0 +1,75 @@
+# -*- coding: binary -*-
+require 'rex/post/meterpreter'
+require 'bindata'
+
+module Rex
+module Post
+module Meterpreter
+module Ui
+
+###
+#
+# Play audio on remote system
+#
+###
+class Console::CommandDispatcher::Stdapi::AudioOutput
+  Klass = Console::CommandDispatcher::Stdapi::AudioOutput
+
+  include Console::CommandDispatcher
+
+  #
+  # List of supported commands.
+  #
+  def commands
+    all = {
+      'play' => 'play an audio file on target system, nothing written on disk'
+    }
+    reqs = {
+      'play' => []
+    }
+
+    filter_commands(all, reqs)
+  end
+
+  #
+  # Name for this dispatcher
+  #
+  def name
+    'Stdapi: Audio Output'
+  end
+
+  def cmd_play(*args)
+    audio_path = nil
+
+    play_start_opts = Rex::Parser::Arguments.new(
+      '-h' => [ false, "Help Banner" ],
+      '-f' => [ true, "The audio file path (warning: will be copied to memory)" ]
+    )
+
+    play_start_opts.parse(args) do |opt, _idx, val|
+      case opt
+      when '-h'
+        print_line('Usage: audio_play [options]\n')
+        print_line('Upload file to targets memory and play it from memory')
+        print_line(play_start_opts.usage)
+        return
+      when '-f'
+        audio_path = val
+      end
+    end
+
+    if audio_path == nil
+      print_line('Please specify a path to an audio file via the -f option or use -h')
+      return
+    end
+
+    print_status('Playing #{audio_path}...')
```

Comment:
This requires double quotes for the expression substitution. It now displays `Playing #{audio_path}...`.

Reply:
Whoops, fixing.